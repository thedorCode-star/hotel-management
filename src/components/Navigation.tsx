'use client';

import { useState, useEffect } from 'react';
import Link from 'next/link';
import { useRouter } from 'next/navigation';

interface User {
  id: string;
  name: string;
  email: string;
  role: string;
}

export default function Navigation() {
  const router = useRouter();
  const [user, setUser] = useState<User | null>(null);
  const [isLoading, setIsLoading] = useState(true);
  const [isMenuOpen, setIsMenuOpen] = useState(false);
  const [hasRefunds, setHasRefunds] = useState(false);

  useEffect(() => {
    const handler = () => checkAuthStatus();
    window.addEventListener('auth-changed', handler);
    checkAuthStatus(); // Initial check
    return () => {
      window.removeEventListener('auth-changed', handler);
    };
  }, []);

  useEffect(() => {
    if (user) {
      checkRefunds();
    }
  }, [user]);

  const checkAuthStatus = async () => {
    try {
      const response = await fetch('/api/auth/me');

      if (response.ok) {
        const data = await response.json();
        setUser(data.user);
      } else {
        localStorage.removeItem('token');
        setUser(null);
      }
    } catch (error) {
      console.error('Auth check error:', error);
      localStorage.removeItem('token');
      setUser(null);
    } finally {
      setIsLoading(false);
    }
  };

  const checkRefunds = async () => {
    try {
      const response = await fetch('/api/refunds');
      if (response.ok) {
        const data = await response.json();
        setHasRefunds(data.refunds && data.refunds.length > 0);
      }
    } catch (error) {
      console.error('Error checking refunds:', error);
    }
  };

  const handleLogout = async () => {
    try {
      await fetch('/api/auth/logout', {
        method: 'POST',
      });
    } catch (error) {
      console.error('Logout error:', error);
    } finally {
      localStorage.removeItem('token'); // Clear any old tokens
      setUser(null);
      router.push('/auth/login');
      window.dispatchEvent(new Event('auth-changed'));
    }
  };

  if (isLoading) {
    return (
      <nav className="bg-white shadow">
        <div className="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8">
          <div className="flex justify-between h-16">
            <div className="flex items-center">
              <div className="animate-pulse bg-gray-200 h-8 w-32 rounded"></div>
            </div>
            <div className="flex items-center">
              <div className="animate-pulse bg-gray-200 h-8 w-24 rounded"></div>
            </div>
          </div>
        </div>
      </nav>
    );
  }

  return (
    <nav className="bg-white/90 backdrop-blur supports-[backdrop-filter]:bg-white/70 border-b border-gray-100">
      <div className="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8">
        <div className="flex justify-between h-16">
          <div className="flex items-center">
            <Link href="/" className="flex-shrink-0 group">
              <h1 className="text-xl font-bold text-blue-600 group-hover:text-blue-700 transition-colors">Hotel Management</h1>
            </Link>
            <div className="hidden md:ml-6 md:flex md:space-x-1">
              <Link
                href={user ? "/dashboard/rooms" : "/rooms"}
                className="px-3 py-2 rounded-md text-sm font-medium text-gray-700 hover:text-blue-600 hover:bg-blue-50/60 transition-colors"
              >
                Rooms
              </Link>
              {user && (
                <>
                  <Link
                    href="/dashboard"
                    className="px-3 py-2 rounded-md text-sm font-medium text-gray-700 hover:text-blue-600 hover:bg-blue-50/60 transition-colors"
                  >
                    Dashboard
                  </Link>
                  <Link
                    href="/dashboard/bookings"
                    className="px-3 py-2 rounded-md text-sm font-medium text-gray-700 hover:text-blue-600 hover:bg-blue-50/60 transition-colors"
                  >
                    Bookings
                  </Link>
                  {/* **NEW: Refunds navigation - only shows when refunds exist */}
                  {hasRefunds && (
                    <Link
                      href="/dashboard/refunds"
                      className="px-3 py-2 rounded-md text-sm font-medium text-gray-700 hover:text-blue-600 hover:bg-blue-50/60 transition-colors flex items-center"
                    >
                      <span className="mr-1">ðŸ’°</span>
                      Refunds
                      <span className="ml-1 bg-red-100 text-red-600 text-xs px-2 py-0.5 rounded-full animate-pulse">
                        {hasRefunds ? '!' : ''}
                      </span>
                    </Link>
                  )}
                  {user.role === 'ADMIN' && (
                    <Link
                      href="/dashboard/admin"
                      className="px-3 py-2 rounded-md text-sm font-medium text-gray-700 hover:text-blue-600 hover:bg-blue-50/60 transition-colors"
                    >
                      Admin
                    </Link>
                  )}
                </>
              )}
            </div>
          </div>

          <div className="flex items-center">
            {user ? (
              <div className="relative">
                <button
                  onClick={() => setIsMenuOpen(!isMenuOpen)}
                  className="flex items-center space-x-2 text-gray-700 hover:text-blue-600 px-3 py-2 rounded-md text-sm font-medium hover:bg-blue-50/60 transition-colors"
                >
                  <span className="truncate max-w-[140px]">{user.name}</span>
                  <svg className="h-4 w-4" fill="none" stroke="currentColor" viewBox="0 0 24 24">
                    <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M19 9l-7 7-7-7" />
                  </svg>
                </button>

                {isMenuOpen && (
                  <div className="absolute right-0 mt-2 w-56 bg-white rounded-xl shadow-lg ring-1 ring-black/5 py-2 z-50">
                    <Link
                      href="/dashboard/user"
                      className="block px-4 py-2 text-sm text-gray-700 hover:bg-gray-50"
                      onClick={() => setIsMenuOpen(false)}
                    >
                      My Profile
                    </Link>
                    <Link
                      href="/dashboard/bookings"
                      className="block px-4 py-2 text-sm text-gray-700 hover:bg-gray-50"
                      onClick={() => setIsMenuOpen(false)}
                    >
                      My Bookings
                    </Link>
                    <button
                      onClick={() => {
                        handleLogout();
                        setIsMenuOpen(false);
                      }}
                      className="block w-full text-left px-4 py-2 text-sm text-gray-700 hover:bg-gray-50"
                    >
                      Sign Out
                    </button>
                  </div>
                )}
              </div>
            ) : (
              <div className="flex space-x-4">
                <Link
                  href="/auth/login"
                  className="text-gray-700 hover:text-blue-600 px-3 py-2 rounded-md text-sm font-medium hover:bg-blue-50/60 transition-colors"
                >
                  Sign In
                </Link>
                <Link
                  href="/auth/register"
                  className="bg-blue-600 text-white px-4 py-2 rounded-md text-sm font-medium hover:bg-blue-700 shadow-sm hover:shadow transition"
                >
                  Sign Up
                </Link>
              </div>
            )}
          </div>

          {/* Mobile menu button */}
          <div className="md:hidden flex items-center">
            <button
              onClick={() => setIsMenuOpen(!isMenuOpen)}
              className="text-gray-700 hover:text-blue-600 p-2"
            >
              <svg className="h-6 w-6" fill="none" stroke="currentColor" viewBox="0 0 24 24">
                <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M4 6h16M4 12h16M4 18h16" />
              </svg>
            </button>
          </div>
        </div>

        {/* Mobile menu */}
        {isMenuOpen && (
          <div className="md:hidden">
            <div className="px-2 pt-2 pb-3 space-y-1 sm:px-3">
              <Link
                href={user ? "/dashboard/rooms" : "/rooms"}
                className="text-gray-900 hover:text-blue-600 block px-3 py-2 rounded-md text-base font-medium"
                onClick={() => setIsMenuOpen(false)}
              >
                Rooms
              </Link>
              {user && (
                <>
                  <Link
                    href="/dashboard"
                    className="text-gray-900 hover:text-blue-600 block px-3 py-2 rounded-md text-base font-medium"
                    onClick={() => setIsMenuOpen(false)}
                  >
                    Dashboard
                  </Link>
                  {user.role === 'ADMIN' && (
                    <Link
                      href="/dashboard/admin"
                      className="text-gray-900 hover:text-blue-600 block px-3 py-2 rounded-md text-base font-medium"
                      onClick={() => setIsMenuOpen(false)}
                    >
                      Admin
                    </Link>
                  )}
                  <Link
                    href="/dashboard/user"
                    className="text-gray-900 hover:text-blue-600 block px-3 py-2 rounded-md text-base font-medium"
                    onClick={() => setIsMenuOpen(false)}
                  >
                    My Profile
                  </Link>
                  <Link
                    href="/dashboard/bookings"
                    className="text-gray-900 hover:text-blue-600 block px-3 py-2 rounded-md text-base font-medium"
                    onClick={() => setIsMenuOpen(false)}
                  >
                    My Bookings
                  </Link>
                  {/* **NEW: Refunds navigation in mobile menu - only shows when refunds exist */}
                  {hasRefunds && (
                    <Link
                      href="/dashboard/refunds"
                      className="text-gray-900 hover:text-blue-600 block px-3 py-2 rounded-md text-base font-medium flex items-center"
                      onClick={() => setIsMenuOpen(false)}
                    >
                      <span className="mr-1">ðŸ’°</span>
                      Refunds
                      <span className="ml-1 bg-red-100 text-red-600 text-xs px-2 py-0.5 rounded-full">
                        !
                      </span>
                    </Link>
                  )}
                  <button
                    onClick={() => {
                      handleLogout();
                      setIsMenuOpen(false);
                    }}
                    className="text-gray-900 hover:text-blue-600 block w-full text-left px-3 py-2 rounded-md text-base font-medium"
                  >
                    Sign Out
                  </button>
                </>
              )}
              {!user && (
                <>
                  <Link
                    href="/auth/login"
                    className="text-gray-900 hover:text-blue-600 block px-3 py-2 rounded-md text-base font-medium"
                    onClick={() => setIsMenuOpen(false)}
                  >
                    Sign In
                  </Link>
                  <Link
                    href="/auth/register"
                    className="text-gray-900 hover:text-blue-600 block px-3 py-2 rounded-md text-base font-medium"
                    onClick={() => setIsMenuOpen(false)}
                  >
                    Sign Up
                  </Link>
                </>
              )}
            </div>
          </div>
        )}
      </div>
    </nav>
  );
} 