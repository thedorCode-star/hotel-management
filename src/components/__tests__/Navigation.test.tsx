import { render, screen, fireEvent, waitFor, act } from '@testing-library/react';
import '@testing-library/jest-dom';
import Navigation from '../Navigation';

// Mock Next.js modules
jest.mock('next/navigation', () => ({
  useRouter: () => ({
    push: jest.fn(),
  }),
}));

jest.mock('next/link', () => {
  return function MockLink({ href, children, ...props }: any) {
    return <a href={href} {...props}>{children}</a>;
  };
});

// Mock localStorage
const localStorageMock = {
  getItem: jest.fn(),
  setItem: jest.fn(),
  removeItem: jest.fn(),
  clear: jest.fn(),
};
Object.defineProperty(window, 'localStorage', {
  value: localStorageMock,
});

// Mock fetch
global.fetch = jest.fn();

describe('Navigation', () => {
  beforeEach(() => {
    jest.clearAllMocks();
    localStorageMock.getItem.mockReturnValue(null);
  });

  it('renders navigation without user', async () => {
    // Mock both API calls - auth check returns no user, refunds check won't be called
    (global.fetch as jest.Mock)
      .mockResolvedValueOnce({
        ok: false,
        status: 401,
        json: async () => ({ error: 'Unauthorized' }),
      });

    await act(async () => {
      render(<Navigation />);
    });
    
    // Wait for the auth check to complete
    await waitFor(() => {
      expect(screen.getByText('Hotel Management')).toBeInTheDocument();
      expect(screen.getByText('Sign In')).toBeInTheDocument();
      expect(screen.getByText('Sign Up')).toBeInTheDocument();
    });
  });

  it('handles logout', async () => {
    const mockUser = {
      id: '1',
      name: 'John Doe',
      email: 'john@example.com',
      role: 'USER',
    };

    // Mock all possible fetch calls with more responses to handle any additional calls
    (global.fetch as jest.Mock)
      .mockResolvedValueOnce({  // First call: /api/auth/me
        ok: true,
        json: async () => ({ user: mockUser }),
      })
      .mockResolvedValueOnce({  // Second call: /api/refunds
        ok: true,
        json: async () => ({ refunds: [] }),
      })
      .mockResolvedValueOnce({  // Third call: /api/auth/logout
        ok: true,
        json: async () => ({ success: true }),
      })
      // Add more mock responses to handle any additional fetch calls
      .mockResolvedValue({
        ok: false,
        status: 401,
        json: async () => ({ error: 'Unauthorized' }),
      });

    localStorageMock.getItem.mockReturnValue('mock-token');

    await act(async () => {
      render(<Navigation />);
    });
    
    // Wait for both API calls to complete and user to be loaded
    await waitFor(() => {
      expect(screen.getByText('John Doe')).toBeInTheDocument();
    });
    
    const userButton = screen.getByText('John Doe');
    fireEvent.click(userButton);
    
    // Use getAllByText and select the first one (desktop menu)
    const signOutButtons = screen.getAllByText('Sign Out');
    const signOutButton = signOutButtons[0];
    fireEvent.click(signOutButton);
    
    // The logout function should be called, but we need to wait for it
    await waitFor(() => {
      expect(localStorageMock.removeItem).toHaveBeenCalledWith('token');
    });
  });

  it('toggles mobile menu', async () => {
    const mockUser = {
      id: '1',
      name: 'John Doe',
      email: 'john@example.com',
      role: 'USER',
    };

    (global.fetch as jest.Mock)
      .mockResolvedValueOnce({
        ok: true,
        json: async () => ({ user: mockUser }),
      })
      .mockResolvedValueOnce({
        ok: true,
        json: async () => ({ refunds: [] }),
      });

    localStorageMock.getItem.mockReturnValue('mock-token');

    await act(async () => {
      render(<Navigation />);
    });
    
    // Wait for both API calls to complete and user to be loaded
    await waitFor(() => {
      expect(screen.getByText('John Doe')).toBeInTheDocument();
    });
    
    // Find the mobile menu button (hamburger menu)
    const mobileMenuButton = screen.getByRole('button', { name: '' });
    fireEvent.click(mobileMenuButton);
    
    // Use getAllByText to handle multiple elements
    const myProfileElements = screen.getAllByText('My Profile');
    const myBookingsElements = screen.getAllByText('My Bookings');
    
    expect(myProfileElements.length).toBeGreaterThan(0);
    expect(myBookingsElements.length).toBeGreaterThan(0);
  });

  it('renders admin navigation for admin user', async () => {
    const mockUser = {
      id: '1',
      name: 'Admin User',
      email: 'admin@example.com',
      role: 'ADMIN',
    };

    (global.fetch as jest.Mock)
      .mockResolvedValueOnce({
        ok: true,
        json: async () => ({ user: mockUser }),
      })
      .mockResolvedValueOnce({
        ok: true,
        json: async () => ({ refunds: [] }),
      });

    localStorageMock.getItem.mockReturnValue('mock-token');

    await act(async () => {
      render(<Navigation />);
    });
    
    // Wait for both API calls to complete and admin user to be loaded
    await waitFor(() => {
      expect(screen.getByText('Admin User')).toBeInTheDocument();
    });
    
    expect(screen.getByText('Admin')).toBeInTheDocument();
  });

  it('renders navigation with user', async () => {
    const mockUser = {
      id: '1',
      name: 'John Doe',
      email: 'john@example.com',
      role: 'USER',
    };

    (global.fetch as jest.Mock)
      .mockResolvedValueOnce({
        ok: true,
        json: async () => ({ user: mockUser }),
      })
      .mockResolvedValueOnce({
        ok: true,
        json: async () => ({ refunds: [] }),
      });

    localStorageMock.getItem.mockReturnValue('mock-token');

    await act(async () => {
      render(<Navigation />);
    });
    
    // Wait for both API calls to complete and user to be loaded
    await waitFor(() => {
      expect(screen.getByText('John Doe')).toBeInTheDocument();
    });
    
    expect(screen.getByText('John Doe')).toBeInTheDocument();
    expect(screen.getByText('Dashboard')).toBeInTheDocument();
  });

  // Snapshot tests
  it('matches snapshot for unauthenticated user', async () => {
    // Mock fetch to return error response with json method
    (global.fetch as jest.Mock).mockResolvedValueOnce({
      ok: false,
      status: 401,
      json: async () => ({ error: 'Unauthorized' }),
    });

    let container: any;
    await act(async () => {
      const result = render(<Navigation />);
      container = result.container;
    });
    
    // Wait for auth check to complete
    await waitFor(() => {
      expect(screen.getByText('Hotel Management')).toBeInTheDocument();
    });
    
    expect(container).toMatchSnapshot();
  });

  it('matches snapshot for authenticated user', async () => {
    const mockUser = {
      id: '1',
      name: 'John Doe',
      email: 'john@example.com',
      role: 'USER',
    };

    (global.fetch as jest.Mock)
      .mockResolvedValueOnce({
        ok: true,
        json: async () => ({ user: mockUser }),
      })
      .mockResolvedValueOnce({
        ok: true,
        json: async () => ({ refunds: [] }),
      });

    localStorageMock.getItem.mockReturnValue('mock-token');

    let container: any;
    await act(async () => {
      const result = render(<Navigation />);
      container = result.container;
    });
    
    // Wait for both API calls to complete and user to be loaded
    await waitFor(() => {
      expect(screen.getByText('John Doe')).toBeInTheDocument();
    });
    
    expect(container).toMatchSnapshot();
  });

  it('matches snapshot for admin user', async () => {
    const mockUser = {
      id: '1',
      name: 'Admin User',
      email: 'admin@example.com',
      role: 'ADMIN',
    };

    (global.fetch as jest.Mock)
      .mockResolvedValueOnce({
        ok: true,
        json: async () => ({ user: mockUser }),
      })
      .mockResolvedValueOnce({
        ok: true,
        json: async () => ({ refunds: [] }),
      });

    localStorageMock.getItem.mockReturnValue('mock-token');

    let container: any;
    await act(async () => {
      const result = render(<Navigation />);
      container = result.container;
    });
    
    // Wait for both API calls to complete and admin user to be loaded
    await waitFor(() => {
      expect(screen.getByText('Admin User')).toBeInTheDocument();
    });
    
    expect(container).toMatchSnapshot();
  });

  it('matches snapshot for loading state', async () => {
    localStorageMock.getItem.mockReturnValue('mock-token');
    // Mock fetch to return a promise that resolves after a delay to simulate loading
    (global.fetch as jest.Mock).mockImplementation(() => 
      new Promise((resolve) => {
        setTimeout(() => {
          resolve({
            ok: false,
            status: 401,
            json: async () => ({ error: 'Unauthorized' }),
          });
        }, 1000); // 1 second delay
      })
    );

    let container: any;
    await act(async () => {
      const result = render(<Navigation />);
      container = result.container;
    });
    
    // The component should be in loading state initially
    expect(container).toMatchSnapshot();
  });

  it('matches snapshot for mobile menu open', async () => {
    const mockUser = {
      id: '1',
      name: 'John Doe',
      email: 'john@example.com',
      role: 'USER',
    };

    (global.fetch as jest.Mock)
      .mockResolvedValueOnce({
        ok: true,
        json: async () => ({ user: mockUser }),
      })
      .mockResolvedValueOnce({
        ok: true,
        json: async () => ({ refunds: [] }),
      });

    localStorageMock.getItem.mockReturnValue('mock-token');

    let container: any;
    await act(async () => {
      const result = render(<Navigation />);
      container = result.container;
    });
    
    // Wait for both API calls to complete and user to be loaded
    await waitFor(() => {
      expect(screen.getByText('John Doe')).toBeInTheDocument();
    });
    
    const mobileMenuButton = screen.getByRole('button', { name: '' });
    fireEvent.click(mobileMenuButton);
    
    expect(container).toMatchSnapshot();
  });
}); 