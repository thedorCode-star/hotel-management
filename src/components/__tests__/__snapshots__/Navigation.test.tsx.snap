// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Navigation matches snapshot for admin user 1`] = `
<div>
  <nav
    class="bg-white/90 backdrop-blur supports-[backdrop-filter]:bg-white/70 border-b border-gray-100"
  >
    <div
      class="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8"
    >
      <div
        class="flex justify-between h-16"
      >
        <div
          class="flex items-center"
        >
          <a
            class="flex-shrink-0 group"
            href="/"
          >
            <h1
              class="text-xl font-bold text-blue-600 group-hover:text-blue-700 transition-colors"
            >
              Hotel Management
            </h1>
          </a>
          <div
            class="hidden md:ml-6 md:flex md:space-x-1"
          >
            <a
              class="px-3 py-2 rounded-md text-sm font-medium text-gray-700 hover:text-blue-600 hover:bg-blue-50/60 transition-colors"
              href="/dashboard/rooms"
            >
              Rooms
            </a>
            <a
              class="px-3 py-2 rounded-md text-sm font-medium text-gray-700 hover:text-blue-600 hover:bg-blue-50/60 transition-colors"
              href="/dashboard"
            >
              Dashboard
            </a>
            <a
              class="px-3 py-2 rounded-md text-sm font-medium text-gray-700 hover:text-blue-600 hover:bg-blue-50/60 transition-colors"
              href="/dashboard/bookings"
            >
              Bookings
            </a>
            <a
              class="px-3 py-2 rounded-md text-sm font-medium text-gray-700 hover:text-blue-600 hover:bg-blue-50/60 transition-colors"
              href="/dashboard/admin"
            >
              Admin
            </a>
          </div>
        </div>
        <div
          class="flex items-center"
        >
          <div
            class="relative"
          >
            <button
              class="flex items-center space-x-2 text-gray-700 hover:text-blue-600 px-3 py-2 rounded-md text-sm font-medium hover:bg-blue-50/60 transition-colors"
            >
              <span
                class="truncate max-w-[140px]"
              >
                Admin User
              </span>
              <svg
                class="h-4 w-4"
                fill="none"
                stroke="currentColor"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 9l-7 7-7-7"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </button>
          </div>
        </div>
        <div
          class="md:hidden flex items-center"
        >
          <button
            class="text-gray-700 hover:text-blue-600 p-2"
          >
            <svg
              class="h-6 w-6"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
            >
              <path
                d="M4 6h16M4 12h16M4 18h16"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </nav>
</div>
`;

exports[`Navigation matches snapshot for authenticated user 1`] = `
<div>
  <nav
    class="bg-white/90 backdrop-blur supports-[backdrop-filter]:bg-white/70 border-b border-gray-100"
  >
    <div
      class="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8"
    >
      <div
        class="flex justify-between h-16"
      >
        <div
          class="flex items-center"
        >
          <a
            class="flex-shrink-0 group"
            href="/"
          >
            <h1
              class="text-xl font-bold text-blue-600 group-hover:text-blue-700 transition-colors"
            >
              Hotel Management
            </h1>
          </a>
          <div
            class="hidden md:ml-6 md:flex md:space-x-1"
          >
            <a
              class="px-3 py-2 rounded-md text-sm font-medium text-gray-700 hover:text-blue-600 hover:bg-blue-50/60 transition-colors"
              href="/dashboard/rooms"
            >
              Rooms
            </a>
            <a
              class="px-3 py-2 rounded-md text-sm font-medium text-gray-700 hover:text-blue-600 hover:bg-blue-50/60 transition-colors"
              href="/dashboard"
            >
              Dashboard
            </a>
            <a
              class="px-3 py-2 rounded-md text-sm font-medium text-gray-700 hover:text-blue-600 hover:bg-blue-50/60 transition-colors"
              href="/dashboard/bookings"
            >
              Bookings
            </a>
          </div>
        </div>
        <div
          class="flex items-center"
        >
          <div
            class="relative"
          >
            <button
              class="flex items-center space-x-2 text-gray-700 hover:text-blue-600 px-3 py-2 rounded-md text-sm font-medium hover:bg-blue-50/60 transition-colors"
            >
              <span
                class="truncate max-w-[140px]"
              >
                John Doe
              </span>
              <svg
                class="h-4 w-4"
                fill="none"
                stroke="currentColor"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 9l-7 7-7-7"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </button>
          </div>
        </div>
        <div
          class="md:hidden flex items-center"
        >
          <button
            class="text-gray-700 hover:text-blue-600 p-2"
          >
            <svg
              class="h-6 w-6"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
            >
              <path
                d="M4 6h16M4 12h16M4 18h16"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </nav>
</div>
`;

exports[`Navigation matches snapshot for loading state 1`] = `
<div>
  <nav
    class="bg-white shadow"
  >
    <div
      class="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8"
    >
      <div
        class="flex justify-between h-16"
      >
        <div
          class="flex items-center"
        >
          <div
            class="animate-pulse bg-gray-200 h-8 w-32 rounded"
          />
        </div>
        <div
          class="flex items-center"
        >
          <div
            class="animate-pulse bg-gray-200 h-8 w-24 rounded"
          />
        </div>
      </div>
    </div>
  </nav>
</div>
`;

exports[`Navigation matches snapshot for mobile menu open 1`] = `
<div>
  <nav
    class="bg-white/90 backdrop-blur supports-[backdrop-filter]:bg-white/70 border-b border-gray-100"
  >
    <div
      class="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8"
    >
      <div
        class="flex justify-between h-16"
      >
        <div
          class="flex items-center"
        >
          <a
            class="flex-shrink-0 group"
            href="/"
          >
            <h1
              class="text-xl font-bold text-blue-600 group-hover:text-blue-700 transition-colors"
            >
              Hotel Management
            </h1>
          </a>
          <div
            class="hidden md:ml-6 md:flex md:space-x-1"
          >
            <a
              class="px-3 py-2 rounded-md text-sm font-medium text-gray-700 hover:text-blue-600 hover:bg-blue-50/60 transition-colors"
              href="/dashboard/rooms"
            >
              Rooms
            </a>
            <a
              class="px-3 py-2 rounded-md text-sm font-medium text-gray-700 hover:text-blue-600 hover:bg-blue-50/60 transition-colors"
              href="/dashboard"
            >
              Dashboard
            </a>
            <a
              class="px-3 py-2 rounded-md text-sm font-medium text-gray-700 hover:text-blue-600 hover:bg-blue-50/60 transition-colors"
              href="/dashboard/bookings"
            >
              Bookings
            </a>
          </div>
        </div>
        <div
          class="flex items-center"
        >
          <div
            class="relative"
          >
            <button
              class="flex items-center space-x-2 text-gray-700 hover:text-blue-600 px-3 py-2 rounded-md text-sm font-medium hover:bg-blue-50/60 transition-colors"
            >
              <span
                class="truncate max-w-[140px]"
              >
                John Doe
              </span>
              <svg
                class="h-4 w-4"
                fill="none"
                stroke="currentColor"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 9l-7 7-7-7"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </button>
            <div
              class="absolute right-0 mt-2 w-56 bg-white rounded-xl shadow-lg ring-1 ring-black/5 py-2 z-50"
            >
              <a
                class="block px-4 py-2 text-sm text-gray-700 hover:bg-gray-50"
                href="/dashboard/user"
              >
                My Profile
              </a>
              <a
                class="block px-4 py-2 text-sm text-gray-700 hover:bg-gray-50"
                href="/dashboard/bookings"
              >
                My Bookings
              </a>
              <button
                class="block w-full text-left px-4 py-2 text-sm text-gray-700 hover:bg-gray-50"
              >
                Sign Out
              </button>
            </div>
          </div>
        </div>
        <div
          class="md:hidden flex items-center"
        >
          <button
            class="text-gray-700 hover:text-blue-600 p-2"
          >
            <svg
              class="h-6 w-6"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
            >
              <path
                d="M4 6h16M4 12h16M4 18h16"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        class="md:hidden"
      >
        <div
          class="px-2 pt-2 pb-3 space-y-1 sm:px-3"
        >
          <a
            class="text-gray-900 hover:text-blue-600 block px-3 py-2 rounded-md text-base font-medium"
            href="/dashboard/rooms"
          >
            Rooms
          </a>
          <a
            class="text-gray-900 hover:text-blue-600 block px-3 py-2 rounded-md text-base font-medium"
            href="/dashboard"
          >
            Dashboard
          </a>
          <a
            class="text-gray-900 hover:text-blue-600 block px-3 py-2 rounded-md text-base font-medium"
            href="/dashboard/user"
          >
            My Profile
          </a>
          <a
            class="text-gray-900 hover:text-blue-600 block px-3 py-2 rounded-md text-base font-medium"
            href="/dashboard/bookings"
          >
            My Bookings
          </a>
          <button
            class="text-gray-900 hover:text-blue-600 block w-full text-left px-3 py-2 rounded-md text-base font-medium"
          >
            Sign Out
          </button>
        </div>
      </div>
    </div>
  </nav>
</div>
`;

exports[`Navigation matches snapshot for unauthenticated user 1`] = `
<div>
  <nav
    class="bg-white/90 backdrop-blur supports-[backdrop-filter]:bg-white/70 border-b border-gray-100"
  >
    <div
      class="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8"
    >
      <div
        class="flex justify-between h-16"
      >
        <div
          class="flex items-center"
        >
          <a
            class="flex-shrink-0 group"
            href="/"
          >
            <h1
              class="text-xl font-bold text-blue-600 group-hover:text-blue-700 transition-colors"
            >
              Hotel Management
            </h1>
          </a>
          <div
            class="hidden md:ml-6 md:flex md:space-x-1"
          >
            <a
              class="px-3 py-2 rounded-md text-sm font-medium text-gray-700 hover:text-blue-600 hover:bg-blue-50/60 transition-colors"
              href="/rooms"
            >
              Rooms
            </a>
          </div>
        </div>
        <div
          class="flex items-center"
        >
          <div
            class="flex space-x-4"
          >
            <a
              class="text-gray-700 hover:text-blue-600 px-3 py-2 rounded-md text-sm font-medium hover:bg-blue-50/60 transition-colors"
              href="/auth/login"
            >
              Sign In
            </a>
            <a
              class="bg-blue-600 text-white px-4 py-2 rounded-md text-sm font-medium hover:bg-blue-700 shadow-sm hover:shadow transition"
              href="/auth/register"
            >
              Sign Up
            </a>
          </div>
        </div>
        <div
          class="md:hidden flex items-center"
        >
          <button
            class="text-gray-700 hover:text-blue-600 p-2"
          >
            <svg
              class="h-6 w-6"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
            >
              <path
                d="M4 6h16M4 12h16M4 18h16"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </nav>
</div>
`;
